org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/ThreadScanTest.java

```diff
@@ -242,7 +241,7 @@ protected synchronized void checkPauseResume(IPausableDevice<?> device, long pau
 
 		device.resume();  // start it going again, non-blocking
 
-		Thread.sleep(10);
+		device.wait(10);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove this call to "wait" or move it into a "while" loop. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2274)